streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamRepartitionTest.java

```diff
@@ -0,0 +1,99 @@
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.kstream.internals;
+
+import org.apache.kafka.common.serialization.IntegerSerializer;
+import org.apache.kafka.common.serialization.Serdes;
+import org.apache.kafka.common.serialization.StringSerializer;
+import org.apache.kafka.streams.KeyValueTimestamp;
+import org.apache.kafka.streams.StreamsBuilder;
+import org.apache.kafka.streams.TestInputTopic;
+import org.apache.kafka.streams.TopologyTestDriver;
+import org.apache.kafka.streams.kstream.KStream;
+import org.apache.kafka.streams.kstream.Repartitioned;
+import org.apache.kafka.streams.processor.StreamPartitioner;
+import org.apache.kafka.test.MockProcessor;
+import org.apache.kafka.test.MockProcessorSupplier;
+import org.apache.kafka.test.StreamsTestUtils;
+import org.easymock.EasyMock;
+import org.easymock.EasyMockRunner;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.Properties;
+
+import static org.easymock.EasyMock.anyInt;
+import static org.easymock.EasyMock.anyString;
+import static org.easymock.EasyMock.eq;
+import static org.easymock.EasyMock.expect;
+import static org.easymock.EasyMock.replay;
+import static org.easymock.EasyMock.verify;
+
+@RunWith(EasyMockRunner.class)
+public class KStreamRepartitionTest {
+    private final String inputTopic = "input-topic";
+
+    private final Properties props = StreamsTestUtils.getStreamsConfig(Serdes.Integer(), Serdes.String());
+
+    private StreamsBuilder builder;
+
+    @Before
+    public void setUp() {
+        builder = new StreamsBuilder();
+    }
+
+    @Test
+    public void shouldInvokePartitionerWhenSet() {
+        final int[] expectedKeys = new int[]{0, 1};
+        final StreamPartitioner<Integer, String> streamPartitionerMock = EasyMock.mock(StreamPartitioner.class);
+
+        expect(streamPartitionerMock.partition(anyString(), eq(0), eq("X0"), anyInt())).andReturn(1).times(1);
+        expect(streamPartitionerMock.partition(anyString(), eq(1), eq("X1"), anyInt())).andReturn(1).times(1);
+        replay(streamPartitionerMock);
+
+        final Repartitioned<Integer, String> repartitioned = Repartitioned
+            .streamPartitioner(streamPartitionerMock)
+            .withName("repartition");
+
+        final KStream<Integer, String> repartitionedStream = builder.<Integer, String>stream(inputTopic)
+            .repartition(repartitioned);
+
+        final MockProcessorSupplier<Integer, String> supplier = new MockProcessorSupplier<>();
+
+        repartitionedStream.process(supplier);
```

Comment:
Using `MockProcessorSupplier` is an old test pattern -- we should use the new `TestOutputTopic` instead.